src/darker/tests/test_main.py

```diff
@@ -775,11 +775,58 @@ def test_modify_file(tmp_path, new_content, expect):
     assert result == expect
 
 
+@pytest.mark.kwparametrize(
+    dict(
+        new_content=TextDocument(lines=['print("foo")']),
+        tty=False,
+        with_pygments=False,
+        expect=('print("foo")\n',),
+    ),
+    dict(
+        new_content=TextDocument(lines=['print("foo")']),
+        tty=False,
+        with_pygments=True,
+        expect=('print("foo")\n',),
+    ),
+    dict(
+        new_content=TextDocument(lines=['print("foo")']),
+        tty=True,
+        with_pygments=False,
+        expect=('print("foo")\n',),
+    ),
+    dict(
+        new_content=TextDocument(lines=['print("foo")']),
+        tty=True,
+        with_pygments=True,
+        expect=(
+            '\x1b[36mprint\x1b[39;49;00m(\x1b[33m"\x1b[39;49;00mfoo'
+            + '\x1b[33m"\x1b[39;49;00m)\n',
+            '\x1b[36mprint\x1b[39;49;00m(\x1b[33m"\x1b[39;49;00m\x1b[33mfoo'
+            + '\x1b[39;49;00m\x1b[33m"\x1b[39;49;00m)\n',
+        ),
+    ),
+)
+def test_print_source(new_content, tty, with_pygments, expect, capsys):
+    """Highlight is applied only if tty, final newline is handled correctly."""
+    with (
+        patch("sys.stdout.isatty", Mock(return_value=tty)),
+        patch(
+            "darker.__main__._import_pygments",
+            Mock(
+                return_value=darker.__main__._import_pygments(),
+                side_effect=None if with_pygments else ImportError(),
+            ),
```

Comment:
Parenthesized context managers are a recent addition to Python syntax. We should support older versions using this old uglier syntax (made even worse by Black):
```suggestion
    with patch("sys.stdout.isatty", Mock(return_value=tty)), patch(
        "darker.__main__._import_pygments",
        Mock(
            return_value=darker.__main__._import_pygments(),
            side_effect=None if with_pygments else ImportError(),
```